corehq/apps/app_manager/views/apps.py

```diff
@@ -496,7 +494,7 @@ def load_app_from_slug(domain, username, slug):
                         app.create_mapping(multimedia, MULTIMEDIA_PREFIX + path)
 
     comment = _("A sample application you can try out in Web Apps")
-    build = make_async_build_v2(app.get_id, app.domain, app.version, allow_prune=False, comment=comment)
+    build = make_build(app, username, allow_prune=False, comment=comment)
     build.is_released = True
     build.save(increment_version=False)
     return build
```

Comment:
I think I'd make a helper function which does this build stuff as well as the necessary pieces of `make_async_build_v2`.  That should end up much simpler.

Reply:
I believe this change needs changing `make_async_build_v2` and then call the extracted method?
but, https://github.com/dimagi/commcare-hq/pull/24860#discussion_r304790702

Reply:
Sorry, what are you suggesting here?

Reply:
Tried, https://github.com/dimagi/commcare-hq/pull/24860/commits/f3b3e82d045eafbb6a79a6436deb00fc8de58076

Reply:
I mean basically take this current code:
```python
# apps.py

    comment = _("A sample application you can try out in Web Apps")
    build = make_build(app, allow_prune=False, comment=comment)
    build.is_released = True
    build.save(increment_version=False)
    return build

# tasks.py
def make_build(app, allow_prune=True, comment=None):
    errors = app.validate_app()
    if not errors:
        return _make_build(app, allow_prune=allow_prune, comment=comment)


def _make_build(app, allow_prune=True, comment=None):
    copy = app.make_build(comment=comment)
    copy.is_auto_generated = allow_prune
    copy.save(increment_version=False)
    return copy
```

and replace it with

```python
# apps.py

    return _build_sample_app(app)

def _build_sample_app(app):
    errors = app.validate_app()
    if not errors:
        comment = _("A sample application you can try out in Web Apps")
        copy = app.make_build(comment=comment)
        copy.is_released = True
        copy.save(increment_version=False)
        return copy
```
(and move the `_make_build` back into the task)

That is, there's actually very little happening in there, but it's hard to see because it's spread across several functions.  If you recombine everything, you can see that it's quite simple.

`allow_prune` is always `False`, so it can be dropped, that means `is_auto_generated` is `False`, which is the default, so that can also be dropped, plus `save` is currently being called twice, which is unecesssary.

Reply:
hmm, this would need to make `is_released` as an argument as well since at places the build needs to be released and at others its not.
Fine to leave this as-is now for this PR? Agree that things can be more cleaner but i want to avoid changing functionality.

Reply:
I approved this PR, so feel free to merge when the build passes.  It sounds like I wasn't adequately explaining myself, so I made a branch off of this PR with the changes I was thinking of:
https://github.com/dimagi/commcare-hq/pull/24913
(no urgency on that though - you can merge this one in and I'll switch the base branch of that one to `master`)